src/main/java/am/ik/yavi/constraint/time/ComparableTemporalConstraintBase.java

```diff
@@ -0,0 +1,92 @@
+package am.ik.yavi.constraint.time;
+
+import am.ik.yavi.constraint.base.ConstraintBase;
+import am.ik.yavi.core.Constraint;
+import am.ik.yavi.core.ConstraintPredicate;
+
+import java.time.temporal.Temporal;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+import static am.ik.yavi.core.NullAs.VALID;
+import static am.ik.yavi.core.ViolationMessage.Default.*;
+
+/**
+ * This is the base class for constraints on Temporal classes that can be compared with
+ * others. Methods in the class require the {@link V} to extend Temporal and implement
+ * Comparable.
+ *
+ * @author Diego Krupitza
+ */
+abstract class ComparableTemporalConstraintBase<T, V extends Temporal & Comparable, C extends Constraint<T, V, C>>
+		extends ConstraintBase<T, V, C> {
+
+	/**
+	 * Is the given temporal before {@code other}
+	 *
+	 * @param other the other temporal that is after
+	 */
+	public C before(V other) {
+		this.predicates().add(ConstraintPredicate.of(x -> x.compareTo(other) <= -1,
+				TEMPORAL_BEFORE, () -> new Object[] {}, VALID));
+		return cast();
+	}
+
+	/**
+	 * Is the given temporal before the supplied {@code other}
+	 *
+	 * @param other the supplier providing the other temporal that is after
+	 */
+	public C before(Supplier<V> other) {
+		return this.before(other.get());
+	}
+
+	/**
+	 * Is the given temporal after {@code other}
+	 *
+	 * @param other the other temporal that is before
+	 */
+	public C after(V other) {
+		this.predicates().add(ConstraintPredicate.of(x -> x.compareTo(other) >= 1,
+				TEMPORAL_AFTER, () -> new Object[] {}, VALID));
+		return cast();
+	}
+
+	/**
+	 * Is the given temporal after the supplied {@code other}
+	 *
+	 * @param other the supplier providing the other temporal that is before
+	 */
+	public C after(Supplier<V> other) {
+		return this.after(other.get());
```

Comment:
`Supplier#get` should be invoked later as follows

```java
	public C after(Supplier<V> other) {
		this.predicates().add(ConstraintPredicate.of(x -> x.compareTo(other.get()) >= 1,
				TEMPORAL_AFTER, () -> new Object[] {}, VALID));
		return cast();
	}
```

Reply:
This is now fixed